core/src/main/java/org/apache/iceberg/BaseReplacePartitions.java

```diff
@@ -20,11 +20,18 @@
 package org.apache.iceberg;
 
 import java.util.List;
+import java.util.Optional;
 import org.apache.iceberg.exceptions.ValidationException;
 import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.util.PartitionSet;
 
 public class BaseReplacePartitions
     extends MergingSnapshotProducer<ReplacePartitions> implements ReplacePartitions {
+
+  private final PartitionSet deletedPartitions = PartitionSet.create(super.getSpecsById());
```

Comment:
If there is a reference to other fields or methods, please do the initialization in the constructor.

Reply:
Sure, done.